examples/developer-tools/src/blocks/factory_base.ts

```diff
@@ -0,0 +1,117 @@
+import * as Blockly from 'blockly';
+
+export const factoryBase = {
+  // Base of new block.
+  init: function() {
+    this.setColour(120);
```

Comment:
Do we care about using best practices? If so this should probably be using a theme. I won't comment on any other "best practices" things until I know if we care or not.

Reply:
Using a theme sounds good, but I'd prefer to do that in a separate PR. I'll also update the toolbox to not just reuse `logic_category` etc. 

I do care about best practices, with the caveat that for some stuff this is just an initial PR to prove the feasibility of using the code generator approach instead of the manual approach used in the old tool, so some of the best practices will be done in later PRs